libs/matdbg/src/ShaderReplacer.cpp

```diff
@@ -300,11 +300,11 @@ void ShaderIndex::encodeShadersToIndices() {
         offset += sizeof(uint32_t);
 
         const char* s = record.decodedShaderText.c_str();
-        size_t length = record.decodedShaderText.length();
-        for (size_t cur = 0; s[cur] != '\0'; cur++) {
+        const size_t length = record.decodedShaderText.length();
+        for (size_t cur = 0; cur < length; cur++) {
             size_t pos = cur;
             size_t len = 0;
-            while (s[cur] != '\n') {
+            while (s[cur] != '\n' && s[cur] != '\0') {
```

Comment:
why look for the `\0` here instead of relying only on `length`, like for the outer loop?

Reply:
Yes they are equivalent, fixed.

Reply:
Actually this was a good catch, this should use `length` rather than `\0` to robust against unicode strings.